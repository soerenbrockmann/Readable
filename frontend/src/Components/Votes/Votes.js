import React from "react";
import "./Votes.css";

const Votes = ({ onUpVote, onDownVote, voteScore }) => {
  return (
    <div className="votes">
      <div className="upvote" onClick={onUpVote}>
        Upvote
      </div>
      {voteScore >= 0 && <div className="votescore">({voteScore})</div>}
      <div className="downvote" onClick={onDownVote}>
        Downvote
      </div>
      {voteScore < 0 && <div className="votescore">({voteScore})</div>}
    </div>
  );
};

export default Votes;
