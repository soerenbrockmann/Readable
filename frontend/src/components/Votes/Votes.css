.votes {
  display: flex;
  flex-direction: row;
}

.votes > div {
  margin-right: 0.5em;
}
