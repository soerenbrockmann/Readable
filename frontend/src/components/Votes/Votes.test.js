import React from "react";
import { shallow } from "enzyme";
import Votes from "./Votes";

describe("<Votes />", () => {
  const votes = shallow(<Votes />);
  it("renders properly", () => {
    expect(votes).toMatchSnapshot();
  });
});
