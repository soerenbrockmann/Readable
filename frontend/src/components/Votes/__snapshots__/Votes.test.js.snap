// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Votes /> renders properly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Votes />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="upvote"
          onClick={undefined}
>
          Upvote
</div>,
        false,
        <div
          className="downvote"
          onClick={undefined}
>
          Downvote
</div>,
        false,
      ],
      "className": "votes",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Upvote",
          "className": "upvote",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Upvote",
        "type": "div",
      },
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Downvote",
          "className": "downvote",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Downvote",
        "type": "div",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="upvote"
            onClick={undefined}
>
            Upvote
</div>,
          false,
          <div
            className="downvote"
            onClick={undefined}
>
            Downvote
</div>,
          false,
        ],
        "className": "votes",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Upvote",
            "className": "upvote",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Upvote",
          "type": "div",
        },
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Downvote",
            "className": "downvote",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Downvote",
          "type": "div",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
